src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -1060,19 +1060,41 @@ public static void AnalyzeTargetHelper(TContext context, IEnumerable<Skimmer<TCo
                         ? Stopwatch.StartNew()
                         : null;
 
+                    if (context.Traces.HasFlag(DefaultTraces.MemoryUsage))
+                    {
+                        using (var currentProcess = Process.GetCurrentProcess())
+                        {
+                            string file = context.TargetUri.LocalPath;
+                            string directory = Path.GetDirectoryName(file);
+                            file = Path.GetFileName(file);
+                            string memoryUsage = $"'{file}' : current process memory usage {currentProcess.WorkingSet64 / 1024 / 1024}MB : '{skimmer.Id}.{skimmer.Name}' : at '{directory}'. Peak memory usage: {currentProcess.PeakWorkingSet64 / 1024 / 1024}MB";
```

Comment:
>memoryUsage [](http://example.com/codeflow?start=35&length=11)

Can you please reply here with a real world example of the new output?

Reply:
yes, I tried to use the same format as the one you added,

[this one:]
BINSKIM : note 'BinaryParsers.dll' : current process memory usage 39MB : 'BA2004.EnableSecureSourceCodeHashing' : at 'C:\GH\binskim\bld\bin\x64_Debug\netcoreapp3.1'. Peak memory usage: 39MB

[the one you added before:]
BINSKIM : note 'BinaryParsers.dll' : elapsed 00:00:00.0263046 : 'BA2004.EnableSecureSourceCodeHashing' : at 'C:\GH\binskim\bld\bin\x64_Debug\netcoreapp3.1'

Reply:
Now that this is changed to log only once the new format is :
`BINSKIM : note Done. 1,386 files scanned, elapsed time 00:00:25.6855079.`
`BINSKIM : note Peak memory usage: 1213MB.`